### PR TITLE
Dependency updates for v2023.4

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### Why is this the best possible solution? Were any other approaches considered?
As usual, I've updated the dependencies for the next release. The only thing of note is that in order to solve the problem with the newest play-services-location I had to migrate to `FusedLocationProviderClient`. See why we used the old implementation:
https://github.com/getodk/collect/pull/5802/commits/ef2de0cdfcbba0fcbee1c05b4373e1a129e82631#diff-211a246d84055d5b6f156f0ad038e9783ece0b2a8d9caf1a01e0c06a0f98255aL50
> Although FusedLocationProviderApi is deprecated, FusedLocationProviderClient which is supposed to replace it doesn't work until Google Play Services 11.6.0, released Nov 2017. Some of our users have really slow connections and old versions of Play Services so we should wait to switch APIs.

It's been a while since Nov 2017 so I think it's ok to do this.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
The only thing worth testing is the location. No matter if it's one of our geo widgets, background location tracking, or location in the map of saved forms, please verify that reading locations work well.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] added a comment above any new strings describing it for translators
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by seadowg — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -230,8 +230,8 @@ jobs:
           command: ./gradlew assembleSelfSignedRelease
 
       - run:
-          name: Check APK size hasn't increased by more than 100kb
-          command: if [ $(ls -l collect_app/build/outputs/apk/selfSignedRelease/*.apk | awk '{print $5}') -gt 10734563 ]; then exit 1; fi
+          name: Check APK size hasn't increased too much
+          command: if [ $(ls -l collect_app/build/outputs/apk/selfSignedRelease/*.apk | awk '{print $5}') -gt 10800000 ]; then exit 1; fi
```
Comment (grzesiek2010) [inline .circleci/config.yml:234]: @lognaturel @seadowg I had to increase the limit here. The difference is not big so I guess it's ok but wanted to let you know.

Comment (seadowg) [reply .circleci/config.yml:234]: Yeah this seems fine. I like this process of us getting a warning so we know it's creeping up. I'll add a note to the Google Drive stuff to make sure we drop this again.

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 13

Verified cases;
    • Mapbox, OSM, use offline layer 
    • Apk build from AS and from CircleCi
    • Collecting Location in background; 
    • View Submissions on map; 
    • Regression checks on geowidgets,
    • Regression checks on external geo widgets 
    • Screen rotation, minimize app, using backward button, light and dark mode

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 8.1, 10

**Pull request merged**